frontend/src/components/Charts/DepthChart/index.tsx

```diff
@@ -0,0 +1,346 @@
+import { AddCircleOutline, RemoveCircleOutline } from '@mui/icons-material';
+import { useTranslation } from "react-i18next";
+import { useHistory } from "react-router-dom"
+import { Order } from "../../../models/Order.model";
+import { LimitList } from "../../../models/Limit.model";
+import RobotAvatar from '../../Robots/RobotAvatar'
+import { amountToString } from "../../../utils/prettyNumbers";
+import currencyDict from '../../../../static/assets/currencies.json';
+import PaymentText from "../../PaymentText";
+import getNivoScheme from "../NivoScheme"
+import median from "../../../utils/match";
+
+interface DepthChartProps {
+  bookLoading: boolean
+  orders: Order[]
+  lastDayPremium: number | undefined
+  currency: number
+  setAppState: (state: object) => void
+  limits: LimitList
+  compact?: boolean
+}
+
+const DepthChart: React.FC<DepthChartProps> = ({ 
+  bookLoading, orders, lastDayPremium, currency, setAppState, limits, compact
+}) => {
+  const { t } = useTranslation()
+  const history = useHistory()
+  const theme = useTheme()
+  const [enrichedOrders, setEnrichedOrders] = useState<Order[]>([])
+  const [series, setSeries] = useState<Serie[]>([])
+  const [xRange, setXRange] = useState<number>(8)
+  const [xType, setXType] = useState<string>("premium")
+  const [currencyCode, setCurrencyCode] = useState<number>(1)
+  const [center, setCenter] = useState<number>(0)
+
+  useEffect(() => {
+    if (Object.keys(limits).length === 0) {
+      fetch('/api/limits/')
+        .then((response) => response.json())
+        .then((data) => {
+          setAppState({ limits: data })
+        })
+    }
+  }, [])
+    
+
+  useEffect(() => {
+    setCurrencyCode(currency === 0 ? 1 : currency)
+  }, [currency])
+
+  useEffect(() => {
+    if (Object.keys(limits).length > 0) {
+      const enriched = orders.map((order) => {
+        // We need to transform all currencies to the same base (ex. USD), we don't have the exchange rate
+        // for EUR -> USD, but we know the rate of both to BTC, so we get advantage of it and apply a
+        // simple rule of three
+        order.base_amount = (order.price * limits[currencyCode].price) / limits[order.currency].price
+        return order
+      })
+      setEnrichedOrders(enriched)
+    }
+  }, [limits, orders, currencyCode])
+
+  useEffect(() => {
+    if (enrichedOrders.length > 0) { 
+      generateSeries()
+    }
+  }, [enrichedOrders, xRange])
+
+  useEffect(() => {
+    if (xType === 'base_amount') { 
+      const prices: number[] = enrichedOrders.map((order) => order?.base_amount || 0)
+      setCenter(~~median(prices))
+      setXRange(1500)
+    } else if (lastDayPremium) {
+      setCenter(lastDayPremium)
+      setXRange(8)
+    }
+  }, [enrichedOrders, xType, lastDayPremium, currencyCode])
+
+  const calculateBtc = (order: Order): number => {
+    const amount = parseInt(order.amount) || order.max_amount
+    return amount / order.price
+  }
```

Comment:
This calculus might potentially be included on the back-end response

Reply:
It can be added to the API. It is still very fluid.

Reply:
I was thinking on creating a separate PR after we merge this one so we don't overcharge it, what do you think?

Reply:
Yes, let's do that on a different PR. Take a look at Model order and the var `order.last_satoshis`, it keeps a snapshot of the size of the order in satoshis when it is created. Is this enough or do we need it to be *realtime* ?

Reply:
It has to be real time, consider everyone creates right their orders and this night Bitcoin pumps up 20%, the amount of BTC you'll be able to buy tomorrow morning is going to be 20% more than the real quantity.

Reply:
I found these 2:

https://github.com/Reckless-Satoshi/robosats/blob/eff58dc91d93fe98a6738e0e3898ed2a476d2ebc/api/models.py#L382
https://github.com/Reckless-Satoshi/robosats/blob/eff58dc91d93fe98a6738e0e3898ed2a476d2ebc/api/models.py#L390

but  the only logic for `t0_satoshis` is this:

https://github.com/Reckless-Satoshi/robosats/blob/eff58dc91d93fe98a6738e0e3898ed2a476d2ebc/api/views.py#L146

Looks like there was an original intention to keep the sats on creation and the sats right now, that's cool, but at some point it got lost and `last_satoshis` was used as `t0_satoshis`.

I like the original idea, I had a look and there are not too many places where `last_satoshis` is being used, do you believe it's a good idea to try a renaming?

Reply:
I think I'll move this conversation to an issue I can assign to myself so it doesn't get lost into this PR